base/src/com/google/idea/blaze/base/dependencies/BlazeQuerySourceToTargetProvider.java

```diff
@@ -201,7 +210,27 @@ private static ImmutableList<TargetInfo> getTargetInfoList(
       return blazeQueryLabelKindParser.getTargets();
     } catch (IOException e) {
       throw new BlazeQuerySourceToTargetException("Failed to get target info list", e);
+    } finally {
+      if (!Registry.get("bazel.sync.keep.query.files").asBoolean()) {
+        queryFile.toFile().delete();
+      }
+    }
+  }
+
+  private static @NotNull Path prepareQueryFile(Project project, String rdepsQuery) throws BlazeQuerySourceToTargetException {
+    Path queryFile = null;
```

Comment:
can we rewrite this function with nio only? ie `Files.createDirectory ` and so on?

Reply:
Done